src/os/ui/layer/vectorlayerui.js

```diff
@@ -659,6 +713,33 @@ os.ui.layer.VectorLayerUICtrl.prototype.onLineDashChange = function(event, value
 };
 
 
+/**
+ * Handles changes to ellipse column mapping
+ *
+ * @param {angular.Scope.Event} event
+ * @param {os.im.mapping.EllipseMappingManager.MappingOptions} value
+ * @protected
+ */
+os.ui.layer.VectorLayerUICtrl.prototype.onEllipseColumnMapping = function(event, value) {
+  event.stopPropagation();
+  const layerNodes = this.getLayerNodes();
+  layerNodes.forEach((node) => {
+    const layer = node.getLayer();
+    const source = /** @type {os.source.Vector} */ (layer.getSource());
+    const sourceId = source.getId();
+
+    const emm = source.getMappingManager(sourceId) || new os.im.mapping.EllipseMappingManager.MappingManager(source);
+
+    emm.setMappingOptions(value);
+    emm.createMappings(value);
+    emm.executeMappings(layer);
+
+    source.setMappingManager(sourceId, emm);
```

Comment:
It looks like there's only one MappingManager per VectorSource, so I'm not sure you need the by-source-id source.getMappingManager(sourceId) when there's just the one.